components/Table/Table.vue

```diff
@@ -2,15 +2,24 @@
   <table>
     <thead>
       <tr>
-        <th v-for="(header, i) in headers" :key="i" :style="textAlign(header.align)">{{header.text}}</th>
+        <th v-for="(header, i) in headers" :key="i" :style="textAlign(header.align)">
+          <slot :name="`header_${header.key}`" :header="header">{{header.text}}</slot>
+        </th>
       </tr>
     </thead>
     <tbody>
-      <tr v-for="(item, j) in items" :key="j">
-        <td v-for="(header,i) in headers" :key="i" :style="textAlign(header.align)">
-          <slot :name="header.key" :item="item">{{ item[header.value] }}</slot>
-        </td>
-      </tr>
+      <template v-for="(item,j) in items">
+        <tr :key="j" @click="toggleItem(j)" :style="hover">
+          <td v-for="header in headers" :key="header.key" :style="textAlign(header.align)">
+            <slot :name="`item_${header.key}`" :item="item">{{ item[header.value] }}</slot>
+          </td>
+        </tr>
+        <tr v-if="expandable && toggle.includes(j)" :key="item.key" :style="hover">
```

Comment:
this one should not have a cursor, if we click nothing is happening. Only the top line should be clickable and so have the hover style.

Also, `item.key` does not exists as a key and like in my previous comment, the key is the same key that you use in the `v-for`. In that case, you can put the same index than on line 12 `:key="j"`

Reply:
I changed the key to `:key="item.date"` (unique date). We shouldn't use a key at the same time.

Reply:
`item.date` is not unique if you change the data with non-unique date ;)
This should depend on the data controlled by the component only.

If you cannot put 2 times `j` then you can prefix is like `item${j}` and `expand${j}` but it has to be related to `j` and not the data from the items

Reply:
Done.